blocks/library/image/index.js

```diff
@@ -19,21 +19,26 @@ registerBlock( 'core/image', {
 		caption: html( 'figcaption' )
 	},
 
-	edit( { attributes, isSelected, setAttributes } ) {
+	edit( { attributes, setAttributes, focus, updateFocus } ) {
 		const { url, alt, caption } = attributes;
+		const focusCaption = () => updateFocus( { editable: 'caption' } );
 
+		/* eslint-disable */
 		return (
 			<figure>
-				<img src={ url } alt={ alt } />
-				{ caption || isSelected ? (
+				<img src={ url } alt={ alt } onClick={ focusCaption } />
+				{ caption || !! focus ? (
 					<Editable
 						tagName="figcaption"
 						placeholder={ wp.i18n.__( 'Write caption…' ) }
 						value={ caption }
+						focus={ focus && focus.editable === 'caption' ? focus : null }
```

Comment:
Do we even need the config here since we only have one focusable element? Wondering if `focusCaption` could be implemented just as:

```
const focusCaption = () => setFocus();
```

And we'd infer caption should be focused merely by a truthy incoming `focus` value?